thrust/testing/unittest/random.h

```diff
@@ -20,58 +20,35 @@ inline unsigned int hash(unsigned int a)
   return a;
 }
 
-template <typename T, typename = void>
-struct generate_random_integer;
-
-template <typename T>
-struct generate_random_integer<
-  T,
-  typename THRUST_NS_QUALIFIER::detail::disable_if<THRUST_NS_QUALIFIER::detail::is_non_bool_arithmetic<T>::value>::type>
-{
-  T operator()(unsigned int i) const
-  {
-    THRUST_NS_QUALIFIER::default_random_engine rng(hash(i));
-
-    return static_cast<T>(rng());
-  }
-};
-
-template <typename T>
-struct generate_random_integer<T, ::cuda::std::enable_if_t<THRUST_NS_QUALIFIER::detail::is_non_bool_integral<T>::value>>
-{
-  T operator()(unsigned int i) const
-  {
-    THRUST_NS_QUALIFIER::default_random_engine rng(hash(i));
-    THRUST_NS_QUALIFIER::uniform_int_distribution<T> dist;
-
-    return static_cast<T>(dist(rng));
-  }
-};
-
 template <typename T>
-struct generate_random_integer<T, typename ::cuda::std::enable_if_t<::cuda::std::is_floating_point<T>::value>>
+struct generate_random_integer
 {
   T operator()(unsigned int i) const
   {
-    T const min = std::numeric_limits<T>::min();
-    T const max = std::numeric_limits<T>::max();
-
     THRUST_NS_QUALIFIER::default_random_engine rng(hash(i));
-    THRUST_NS_QUALIFIER::uniform_real_distribution<T> dist(min, max);
-
-    return static_cast<T>(dist(rng));
-  }
-};
-
-template <>
-struct generate_random_integer<bool>
-{
-  bool operator()(unsigned int i) const
-  {
-    THRUST_NS_QUALIFIER::default_random_engine rng(hash(i));
-    THRUST_NS_QUALIFIER::uniform_int_distribution<unsigned int> dist(0, 1);
-
-    return dist(rng) == 1;
+    if constexpr (::cuda::std::is_same_v<T, bool>)
+    {
+      THRUST_NS_QUALIFIER::uniform_int_distribution<unsigned int> dist(0, 1);
+      return dist(rng) == 1;
+    }
+    else if constexpr (::cuda::std::is_integral_v<T>)
+    {
+      T const min = std::numeric_limits<T>::min();
+      T const max = std::numeric_limits<T>::max();
```

Comment:
Nitpick: This should probably be
```suggestion
      T const max = cuda::std::numeric_limits<T>::max();
```